### ISSUE TITLE
[API Proposal]:

### ISSUE THREAD

**Issue by Lanayx** [dotnet/runtime#109591]
### Background and motivation

I suggest adding `StringBuilder` overload in [WebUtility](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Net/WebUtility.cs) similar to `TextWriter` one

```csharp
public static void HtmlEncode(string? value, StringBuilder output)
```
Motivation: in Oxpecker framework (which is currently one of the top frameworks at [Techempower](https://www.techempower.com/benchmarks/#section=test&runid=1300aee6-f9c9-42a2-8d17-252ba597202f&hw=ph&test=fortune&p=zik0zi-zik0zj-zik0zj-zik0zj-zik0zj-1kv) benchmarks. I'm using `StringBuilder` for ViewEngine rather than `TextWriter` because it gives about 10% performance increase for rendering to string. One downside is that `HtmlEncode` method doesn't allow for StringBuilder, so I'm [allocating strings each time encoding is required](https://github.com/Lanayx/Oxpecker/blob/6f8ef1287d0178894480b0bb830939387163b363/src/Oxpecker.ViewEngine/Builder.fs#L35).
If overload is added, it can behave faster than `TextWriter` since no intermediate `ValueStringBuilder` instances [are requried](https://github.com/dotnet/runtime/blob/6d23ef4d68bbcdb38fdc22218d1073c5083ac6a1/src/libraries/System.Private.CoreLib/src/System/Net/WebUtility.cs#L67).

### API Proposal

```csharp
public static class WebUtility
{
    public static void HtmlEncode(string? value, StringBuilder output)
}
```

### API Usage

```csharp
var sb = new StringBuilder();
WebUtility.HtmlEncode("myString", sb);
```

**Comment by timcassell** [comment]
https://learn.microsoft.com/en-us/dotnet/api/system.io.stringwriter.-ctor?view=net-8.0#system-io-stringwriter-ctor(system-text-stringbuilder)

**Comment by Lanayx** [comment]
@timcassel Yes, I'm aware of this type, but as I said - using it gives 10% overhead in my benchmarks compared to straight StringBuilder usage. With implemented StringBuilder overload performance difference will be even bigger (due to aforementioned possibility of getting rid of intermediate string builder in implementation)

**Comment by julealgon** [comment]
@Lanayx please add a proper title to the issue so people are not forced to open the link to see what it is about.

**Comment by Lanayx** [comment]
Thank you for noticing, done

**Comment by ManickaP** [comment]
I'm not conceptually opposed to this. However, it does increase the API surface. So I'd like to see if there are more people who could benefit from this addition before we add it.

**Comment by MihaZupan** [comment]
While I get it can make usage eaiser, we don't tend to expose `StringBuilder` overloads for such encoding helpers, having a `TextWriter` overload is already very untypical.
If we were to add APIs here, they may be span-based instead, e.g. something like https://github.com/dotnet/runtime/issues/40603#issuecomment-1930539476.

**Comment by ManickaP** [comment]
> If we were to add APIs here, they may be span-based instead, e.g. something like https://github.com/dotnet/runtime/issues/40603#issuecomment-1930539476.

Looks good!

**Comment by Lanayx** [comment]
> If we were to add APIs here, they may be span-based instead, e.g. something like https://github.com/dotnet/runtime/issues/40603#issuecomment-1930539476.

Using spans for these APIs is not developer-friendly in my view, since it's not clear what is the size of buffer that should be allocated, so it's should be up to algorithm and be based on analyzed symbols, rather than developer just allocate buffer 4 times larger and hoping that it will be sifficient. StringBuilder overload will be much more convenient.
